WordPress/openverse
# Fix Vale by passing files to process instead of a glob to ignore

## Fixes



Fixes #3764 by @sarayourfriend 

## Description



Something with Vale's globbing logic would throw an error if a file wasn't found, typically in the case of broken/invalid symlinks. This was true even if the file wasn't one which would be processed by Vale, i.e, non-markdown files.

To fix this, instead of telling vale which files to ignore, I've written a script in our justfile to:

1. List all MD and MDX files in the repository
2. Filter out the directories we do not want to lint
3. Strip the absolute path from the beginning of the files
4. Replace newlines with spaces

This is then passed as the argument to Vale. 

### Concerns and considerations

Some potential pitfalls or annoyances with this approach:

- It doesn't actually fix whatever the underlying problem was
- The list of filenames to pass to Vale is printed to the terminal and clutters our precommit output
- If the repo or the list of files we want to lint grows considerably, we could run into bash argument length limits

## Testing Instructions


1. Edit a markdown file which should be linted by Vale, for example adding "OpenVerse" to one of our readmes.
2. Also edit a file which _should not_ be linted by vale, like our changelogs, to contain "OpenVerse" as well.
3. Manually run precommit with `pre-commit run --all-files`. Confirm the correct Vale errors for the changes you made appear.
4. **Bonus:** `cd ./vale && j debug`. Manually review the list of files being linted and check if anything you think should be there is missing, or if files that shouldn't be linted are.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
